### PR TITLE
subsetDT bugfix

### PR DESCRIPTION
**Pull request by CeresBarros**
Shouldn't impact anyone and I don't *think* it breaks cached Biomass_borealDataPrep::init

### CODE REVIEW

**Comment by achubaty** [comment]
can you please summarize the bug? why can't we use `vapply`?

**Comment by CeresBarros** [comment]
I think `vapply` was not allowing to return an empty list/vector (when there are no errors), which `lapply` does.

**Comment by achubaty** [comment]
is an empty logical a valid output for a logical test?

**Comment by CeresBarros** [comment]
pretty sure I tried and it didn't work. If there are no errors the result is literally `NULL` not a `logical(0)`. I think the `lapply` is fine here, honestly, because there's literally no outout (the only reason there's an `out <- ...` is so that it doesn't print a `list` of `NULL`s

**Comment by achubaty** [comment]
hmm... i think i would prefer the test be fixed (eg use `isTRUE()` or `isFALSE()`, which deal with NULL values). non-type-safe functions make me a sad panda.

**Comment by CeresBarros** [comment]
But the test is _inside_ the `vapply`, not outside... basically, you're saying that I should have the `vapply` output a vector and _then_ test it, but I want the test to be inside the `vapply` so that I can print the messages easily.

**Comment by achubaty** [comment]
no, not necessarily. i'm saying that the output of the function should be tweaked to ensure it's a `logical` of length one everytime.

File: R/cohorts.R [hunk 1]
```diff
@@ -1117,7 +1117,13 @@ subsetDT <- function(DT, by, doSubset = TRUE, indices = FALSE) {
       } else {
         DT <- a$lineNum
       }
+    } else {
+      if (isTRUE(indices))
+        DT <- 1:nrow(DT)
```
Comment (achubaty) [inline R/cohorts.R:1122]: i don't think this is quite right -- we want to add the indices as a column, not replace the data.table, correct?

Comment (CeresBarros) [reply R/cohorts.R:1122]: when `isTRUE(indices)` it's supposed to return a vector indices only, see lines above where:
```
if (isFALSE(indices)) {
        DT <- DT[a$lineNum]
      } else {
      } else {
        DT <- a$lineNum
      }
      }
```

**Comment by achubaty** [comment]
ah yes, sorry.

also note: if subsetting it checks `indices` using `isFALSE` and of not subsetting it uses `isTRUE`. easier if both can use the same logic

**Comment by CeresBarros** [comment]
I get that, but isn't using ` !isFALSE()` and bit silly when `isTRUE` exists?

**Comment by achubaty** [comment]
no, i'm not saynig to use `!ifFALSE()`. I'm saying that the if/else blocks should be constructed like this:

```r
if (isTRUE(thing)) {
   doTheThing()
} else {
  doAnotherThing()
}

if (isTRUE(thing)) {
   doTheThing()
} else {
  doAnotherThing()
}
```

instead of :
```r
if (isTRUE(thing)) {
   doTheThing()
} else {
  doAnotherThing()
}

if (isFALSE(thing)) {
  doAnotherThing()
} else {
   doTheThing()
}
```

**Comment by CeresBarros** [comment]
okay, honestly I'm not sure what you want anymore, so I've reverted the `isTRUE` -> `!isFALSE`.

**Comment by CeresBarros** [comment]
commit 6689e8a0a8f17ae9b1afc72a140fb4d2a7692f54 should actually be it's own PR :/ it fixes double "_" in species layers names

**Comment by achubaty** [comment]
> okay, honestly I'm not sure what you want anymore, so I've reverted the `isTRUE` -> `!isFALSE`.

`if (TRUE) thing else otherThing` is clearer than `if (FALSE) otherThing else thing`, even if they are the same, especially when the preceeding similar if/else block is constructed using the first (clearer) approach.

(this is nitpicky perhaps, and not related directly to your PR. But since you were already modifying that code I thought I'd point it out)

**Review by achubaty — APPROVED** [review]

**Pull request merged**